netket/operator/_sumoperators.py

```diff
@@ -40,6 +40,21 @@ def __init__(
             coefficients: A coefficient for each ContinuousOperator object
             dtype: Data type of the matrix elements. Defaults to `np.float64`
         """
+        if len(operators) != len(coefficients):
+            raise AssertionError("Each operator needs a coefficient")
+
+        new_operators = []
+        new_coeffs = []
+        for op, c in zip(operators, coefficients):
+            if isinstance(op, SumOperator):
+                new_operators = new_operators + op._ops
+                new_coeffs = new_coeffs + list(c * jnp.array(op._pack_arguments()[0]))
```

Comment:
```suggestion
                new_coeffs = new_coeffs + list(c * op._coeff)
```

I think you forgot to update this as well